packages/plugin-hardhat/src/index.ts

```diff
@@ -1,29 +1,37 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
+// TODO figure out what is the newer, TASK_COMPILE_SOLIDITY and TASK_COMPILE_SOLIDITY_COMPILE
```

Comment:
These are Hardhat internal subtasks, where Hardhat calls TASK_COMPILE_SOLIDITY -> TASK_COMPILE_SOLIDITY_COMPILE_JOBS -> TASK_COMPILE_SOLIDITY_COMPILE_JOB ->  TASK_COMPILE_SOLIDITY_COMPILE.  And only certain parts of those need to be overridden in our plugin.  See https://github.com/NomicFoundation/hardhat/blob/09951afd78e49465f7949478919c9217a12e8338/packages/hardhat-core/src/builtin-tasks/compile.ts#L1366 for the v2 internal subtasks.

We may need to find the equivalents in v3.